pkg/workloads/cortex/lib/type/predictor.py

```diff
@@ -21,24 +21,29 @@
 from cortex.lib.log import refresh_logger, cx_logger
 from cortex.lib.exceptions import CortexException, UserException, UserRuntimeException
 from cortex.lib.type.model import Model, get_signature_keys, get_name_signature_pairs
+from cortex import consts
 
 
 class Predictor:
-    def __init__(self, provider, cache_dir, **kwargs):
+    def __init__(self, provider, model_dir, cache_dir, **kwargs):
         self.provider = provider
         self.type = kwargs["type"]
         self.path = kwargs["path"]
         self.python_path = kwargs.get("python_path")
         self.config = kwargs.get("config", {})
         self.env = kwargs.get("env")
 
+        self.model_dir = model_dir
         self.models = []
         if kwargs.get("model"):
```

Comment:
Check if we can remove this section (I pushed the change already)

Reply:
It's used in one place: `self._compute_model_basepath`. I didn't remove it because `self._compute_model_basepath` method also uses another attribute of the class, so it would be too much to pass 2 extra arguments to the function when they can be kept as attributes instead. The reason is that you wouldn't remove just `self.model_dir` but also the other attribute `self.type` from `self._compute_model_basepath`.

What do you think?

Reply:
I didn't follow exactly what you are saying, let me know when is a good time to discuss it!